src/pages/index.js

```diff
@@ -24,24 +24,189 @@ const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
 
-/* Funtions */
+/* Api Endpoints */
+const userProfile = new UserInfo(".profile__title", ".profile__description");
+
+/* Form Validators */
+const profileEditForm = profileEditPopup.querySelector(".popup__form");
+const addCardForm = addCardPopup.querySelector(".popup__form");
+
+const profileEditFormValidator = new FormValidator(settings, profileEditForm);
+const addCardFormValidator = new FormValidator(settings, addCardForm);
+
+profileEditFormValidator.enableValidation();
+addCardFormValidator.enableValidation();
+
+/* Popup Instances */
+const deleteCard = new PopupWithFormSubmit("#trash-popup");
+const imagePopup = new PopupWithImage("#image-popup");
+const avatarPopup = new PopupWithForm("#avatar-edit-popup", handleAvatarSubmit);
+const profilePopup = new PopupWithForm(
+  "#profile-edit-popup",
+  handleProfileEditSubmit
+);
+const cardPopup = new PopupWithForm("#add-card-popup", handleAddCardFormSubmit);
+
+deleteCard.setEventListeners();
+imagePopup.setEventListeners();
+avatarPopup.setEventListeners();
+profilePopup.setEventListeners();
+cardPopup.setEventListeners();
+
+/* Get Initial Cards */
+let cardList;
+api.getInitialCards().then((initialCards) => {
+  console.log("intialCards", initialCards);
+  cardList = new Section(
+    {
+      items: initialCards,
+      renderer(item) {
+        const cardElement = renderCard(item);
+        this._container.appendChild(cardElement);
+      },
+    },
+    ".cards__list"
+  );
+  cardList.renderItems();
+
+  /* Get User Info */
+  api
+    .getUserInfo()
+    .then((userInfo) => {
+      console.log("User information retrieved successfully:", userInfo);
+      if (userInfo) {
+        document.querySelector(".profile__image").src = userInfo.avatar;
+      } else {
+        console.error("Invalid user information received");
+      }
+    })
+    .catch((error) => {
+      console.error("Error getting user information:", error);
+    });
+});
+
+/* Profile Image Click Handler */
+const profileImage = document.querySelector(".profile__image");
+profileImage.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+/* Handlers */
 function handleImageClick(cardData) {
   imagePopup.open(cardData);
 }
+
 function handleProfileEditSubmit(inputValues) {
+  const updatedData = {
+    name: inputValues.name,
+    about: inputValues.about,
+  };
+
+  // Change button text to "Saving..."
+  profilePopup.setLoadingState(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```